CoffeeChat/Controller/GroupsViewController.swift

```diff
@@ -82,6 +150,84 @@ class GroupsViewController: UIViewController {
         }
     }
 
+    override func viewDidAppear(_ animated: Bool) {
+        let clearColor = UIColor(displayP3Red: 1, green: 1, blue: 1, alpha: 0)
+
+        let topLayer = CAGradientLayer()
+        topLayer.frame = topFadeView.bounds
+        topLayer.colors = [UIColor.backgroundWhite.cgColor, clearColor.cgColor]
+        topLayer.locations = [0.0, 1.0]
+        topFadeView.layer.insertSublayer(topLayer, at: 0)
+
+        let bottomLayer = CAGradientLayer()
+        bottomLayer.frame = bottomFadeView.bounds
+        bottomLayer.colors = [clearColor.cgColor, UIColor.backgroundWhite.cgColor]
+        bottomLayer.locations = [0.0, 1.0]
+        bottomFadeView.layer.insertSublayer(bottomLayer, at: 0)
+    }
+
+    // MARK: - Search Bar
+    /** Updates the search bar text based on currently selected Groups. */
+    private func updateSearchBarText() {
+        if selectedGroups.isEmpty {
+            searchBar.text = ""
+            return
+        }
+
+        /** Number of characters before list is considered "too long" and shortened to "..." */
+        let tooLong = 20
+        let wordLong = 15
+
+        let listString = (selectedGroups.map { $0.name } ).joined(separator: ", ")
```

Comment:
`let listString = (selectedGroups.map { $0.name }).joined(separator: ", ")` -- Just an extra space here so remove that!